components/org.wso2.carbon.identity.api.server.branding.preference.management/org.wso2.carbon.identity.api.server.branding.preference.management.v1/src/main/java/org/wso2/carbon/identity/api/server/branding/preference/management/v1/core/BrandingAIPreferenceManagementService.java

```
@@ -0,0 +1,213 @@
+            errorResponseBuilder.withDescription(error.getServerMessage());
+            return new APIError(status, errorResponseBuilder.build());
+        }
+        return new APIError(Response.Status.BAD_REQUEST, errorResponseBuilder.build());
+    }
+
+    private APIError handleServerException(AIServerException error) {
+
+        LOG.error("Server error occurred while generating branding preference.", error);
+        ErrorResponse.Builder errorResponseBuilder = new ErrorResponse.Builder()
+                .withCode(error.getErrorCode())
+                .withMessage(error.getMessage());
+        if (error.getServerMessage() != null) {
+            Response.Status status = Response.Status.fromStatusCode(error.getServerStatusCode());
+            errorResponseBuilder.withDescription(error.getServerMessage());
+            return new APIError(status, errorResponseBuilder.build());
+        }
+        return new APIError(Response.Status.INTERNAL_SERVER_ERROR, errorResponseBuilder.build());
+    }
+
+    private static Map<String, Object> convertObjectToMap(Object object) {
+
+        if (object instanceof Map) {
+            Map<String, Object> map = new HashMap<>();
+            for (Map.Entry<String, Object> entry : ((Map<String, Object>) object).entrySet()) {
+                String key = entry.getKey();
+                Object value = entry.getValue();
+
+                if (value == null) {
+                    map.put(key, "");
+                } else if (value instanceof Map) {
+                    map.put(key, convertObjectToMap(value));
+                } else if (value instanceof List) {
+                    map.put(key, convertListToArray((List<?>) value));
+                } else {
+                    map.put(key, value);
+                }
+            }
+            return map;
+        }
+        LOG.warn("Object is not an instance of Map. Returning an empty map.");
```

lets throw a server exception here.